chain/pra/client.go

```diff
@@ -41,6 +41,15 @@ type SubstrateClient interface {
 	GetBlockHash(blockNumber uint64) (SubstrateHash, error)
 	GetStorageRaw(key SubstrateStorageKey, blockHash SubstrateHash) (*SubstrateStorageDataRaw, error)
 	GetBlockHashLatest() (SubstrateHash, error)
+	GetReadProof(key SubstrateStorageKey, hash SubstrateHash) (ReadProof, error)
+}
+
+type PraClient interface {
```

Comment:
No need to create a new interface. We have enough interfaces here.
`IsSendMessageEvent(e EventEVMLog) bool` -> put into the BMCContract if it does not exist

`CreateSystemEventsStorageKey(hash SubstrateHash) (SubstrateStorageKey, error)` -> put into the SubstrateClient if it does not exist

Reply:
use this command to re-generate mocks it is needed.
`mockery --name NameOfInterface`.
P/S: must cd into the pra folder before calling the above command